src/discord.ts

```diff
@@ -49,11 +55,59 @@ export async function notifyPayment(
               inline: true
             }
           ],
-          timestamp: new Date(block.timestamp * 1000).toISOString()
+          timestamp: new Date(blockTimestamp * 1000).toISOString()
         }
       ]
     })
   });
 
   return;
 }
+
+export async function sendExpirationNotification(
+  categorizedSpaces: CategorizedSpaces
+): Promise<void> {
+  if (!DISCORD_EXPIRATION_WEBHOOK_URL) return;
+  const { expired, expiring } = categorizedSpaces;
+
+  try {
+    const sections: string[] = ['💸 **Snapshot Pro expirations**'];
+
+    if (expired.length > 0) {
+      sections.push('\n**💀 Expired (within last 7 days)**');
+      sections.push(
+        ...expired.map(
+          space =>
+            `❌ **[${space.id}](${SNAPSHOT_BASE_URL}/#/s:${space.id}/settings/billing)** — <t:${space.expiration}:R> (<t:${space.expiration}:f>)`
+        )
+      );
+    }
+
+    if (expiring.length > 0) {
+      sections.push('\n**⏰ Expiring soon (next 7 days)**');
+      sections.push(
+        ...expiring.map(
+          space =>
+            `⚠️ **[${space.id}](${SNAPSHOT_BASE_URL}/#/s:${space.id}/settings/billing)** — <t:${space.expiration}:R> (<t:${space.expiration}:f>)`
+        )
+      );
+    }
+
+    const response = await fetch(DISCORD_EXPIRATION_WEBHOOK_URL, {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ content: sections.join('\n') })
+    });
+    if (!response.ok)
+      throw new Error(
+        `Discord webhook responded with status ${response.status}`
+      );
+    console.log(
+      `Sent notification for ${expired.length + expiring.length} spaces (${
+        expired.length
+      } expired, ${expiring.length} expiring)`
+    );
+  } catch (error) {
+    console.error('Failed to send expiration notification:', error);
+  }
```

Comment:
```suggestion
  } catch (err) {
    console.error('Failed to send expiration notification:', err);
  }
```

For the future enforcement 😆